File: fight-game-frontend/src/message/Message.js
export class Message {

    msgType = "";
    arrowEvent;

    static create() {
        return new Message();
    }
}

export const CREATE_FIGHT = "CREATE_FIGHT";
export const FIGHT_CREATED = "FIGHT_CREATED";
export const JOIN_FIGHT = "JOIN_FIGHT";
export const JOINED_TO_FIGHT = "JOINED_TO_FIGHT";
export const PLAYER_STATE = "PLAYER_STATE";
export const FIGHT_UPDATE = "FIGHT_UPDATE";


export const PLAYER_ID = "player_id";
export const FIGHT_ID = "fight_id";

File: fight-game-backend/src/main/java/budzko/backend/game/fight/app/messaging/message/Message.java
package budzko.backend.game.fight.app.messaging.message;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;

import java.util.Map;

@Data
public class Message {

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private Map<String, Boolean> arrowEvent;
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private Map<String, String> mouseEvent;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String fightId;
    private MessageType msgType;
    @JsonIgnore
    private String userId;

    public static Message create() {
        return new Message();
    }
}
